src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -343,21 +343,45 @@ export function isValidMethodSignature(symbol: vscode.DocumentSymbol): boolean {
     const lambdaContextType = 'ILambdaContext '
 
     if (symbol.kind === vscode.SymbolKind.Method) {
+        // public void methodName(Foo<Bar, Baz> x, ILambdaContext y) -> (Foo<Bar, Baz> x, ILambdaContext y)
         const parametersArr = parametersRegExp.exec(symbol.name)
         // reject if there are no parameters
         if (!parametersArr) {
             return false
         }
-        // don't worry about removing parenthesis
-        const individualParams = parametersArr[0].split(',')
-        if (individualParams.length === 1 || individualParams[1].valueOf().includes(lambdaContextType)) {
+        // remove generics from parameter string so we can do a predictable split on comma
+        const strippedStr = stripGenericsFromParams(parametersArr[0])
+        const individualParams = strippedStr.split(',')
+        if (
+            individualParams.length === 1 ||
+            individualParams[1]
+                .valueOf()
+                .trimLeft()
+                .startsWith(lambdaContextType)
+        ) {
             return true
         }
     }
 
     return false
 }
 
+/**
+ * Strips any generics from a string in order to ensure predictable commas for a string of parameters.
+ * e.g.: `'(Foo<Bar, Baz> x, ILambdaContext y)' -> '(Foo x, ILambdaContext y)'`
+ * Implements a fairly rough English-centric approximation of the C# identifier spec:
+ * * can start with a letter, underscore, or @ sign
+ * * all other characters are letters, numbers, underscores, or periods
+ *
+ * Actual spec: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#identifiers
+ * @param input String to remove generics from
+ */
+function stripGenericsFromParams(input: string): string {
```

Comment:
This could be considered a standalone utility method that can be tested in an easy fashion. Up to you.

Reply:
I'm adding tests to some of the higher level functions that can trickle-down test this. I don't see a need to export this just for testing purposes.